tests/unit/environments/GaussianCheckerboard.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2016, Shogun-Toolbox e.V. <shogun-team@shogun-toolbox.org>
+ * All rights reserved.
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  1. Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *
+ *  3. Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived from
+ *     this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Authors: 2016 MikeLing, Viktor Gal, Sergey Lisitsyn, Heiko Strathmann
+ */
+
+#ifndef BINARY_LABEL_DATA_HPP
+#define BINARY_LABEL_DATA_HPP
+
+#include <shogun/features/DataGenerator.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/labels/BinaryLabels.h>
+
+using namespace shogun;
+
+class GaussianCheckerboard
+{
+public:
+	GaussianCheckerboard(
+	    const int32_t num_samples, const int32_t num_dims,
+	    const int32_t num_labels)
+	{
+		SGMatrix<float64_t> data = CDataGenerator::generate_gaussians(
+		    num_samples, num_dims, num_labels);
+		CDenseFeatures<float64_t> features(data);
+
+		SGVector<index_t> train_idx(num_samples), test_idx(num_samples);
+		SGVector<float64_t> labels(num_samples);
+		for (index_t i = 0, j = 0; i < data.num_cols; ++i)
```

Comment:
@MikeLing no. first of all `num_dim` should be anything as the Gaussian will give you the right dimension (if you set it right).

and `GaussianCheckerboard` should be able to generate a multiclass problem... so `num_dim > 2` should be supported... it's only that you need to create and sample `num_dim` amount of gaussians, and each of them will have a different label... so in case of `num_dim=2` and `num_labels=10` it would look like a checker board (this is how the means of the gaussians should be set) where each 'square' has a different label